src/backend/marsha/core/serializers.py

```diff
@@ -299,6 +302,23 @@ class Meta:  # noqa
     thumbnail = ThumbnailSerializer(read_only=True, allow_null=True)
     urls = serializers.SerializerMethodField()
     is_ready_to_show = serializers.BooleanField(read_only=True)
+    has_transcript = serializers.SerializerMethodField()
+
+    def get_has_transcript(self, obj):
+        """Compute if should_use_subtitle_as_transcript behavior is disabled.
+
+        Parameters
+        ----------
+        obj : Type[models.Video]
+            The video that we want to serialize
+
+        Returns
+        -------
+        Boolean
+            If there is at least one transcript ready to be shown the method will return True.
+            Returns False otherwise.
+        """
+        return obj.timedtexttracks.filter(mode="ts").exists()
```

Comment:
Should you also filter on `uploaded_on` (and/or `upload_state` I'm not sure anymore) to exclude timed text tracks that are not ready to be shown as explained in the docstring? Add tests for that?

Reply:
Done :+1: